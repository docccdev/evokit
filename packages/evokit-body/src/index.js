import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
