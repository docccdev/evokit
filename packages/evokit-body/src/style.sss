@import './mixins.sss'

@custom-media --ek-body-media-small only screen and (min-width: 480px)
@custom-media --ek-body-media-medium only screen and (min-width: 768px)
@custom-media --ek-body-media-large only screen and (min-width: 960px)
@custom-media --ek-body-media-wide only screen and (min-width: 1200px)
@custom-media --ek-body-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-body-font-family: system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif
    --ek-body-font-size: 13px
    --ek-body-font-weight: 400
    --ek-body-line-height: 1em
    --ek-body-min-width: 320px
    --ek-body-max-width: 1240px
    // prop 'body-indent'
    --ek-body-indent-xxs: 5px
    --ek-body-indent-xs: 10px
    --ek-body-indent-s: 15px
    --ek-body-indent-m: 20px
    --ek-body-indent-l: 25px
    --ek-body-indent-xl: 30px
    --ek-body-indent-xxl: 35px
    // prop 'body-size'
    --ek-body-max-width-xl: 1440px
    --ek-body-max-width-xxl: 1640px
    --ek-body-max-width-xxxl: 1840px

.body
    display: block
    font-family: var(--ek-body-font-family)
    font-size: var(--ek-body-font-size)
    line-height: var(--ek-body-line-height)
    font-weight: var(--ek-body-font-weight)
    min-width: var(--ek-body-min-width)
    margin: 0
    text-size-adjust: 100%
    -webkit-font-smoothing: subpixel-antialiased

.body__section
    display: block
    box-sizing: border-box
    margin-left: auto
    margin-right: auto
    min-width: var(--ek-body-min-width)
    max-width: var(--ek-body-max-width)

/**
 * Body display:
 *
 *  - body_display_block
 *  - body_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body_display_$(value)
        display: $value

/**
 * Section display:
 *
 *  - body__section_display_block
 *  - body__section_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body__section_display_$(value)
        display: $value

/**
 * body-size:
 *
 *  - body_size_[xl|xxl|xxxl]
 *
 **/

@each $value in (xl, xxl, xxxl)
    @mixin media .body_size_$(value)
        .body__section
            max-width: var(--ek-body-max-width-$(value))

/**
 * body-indent:
 *
 *  - body_indent_[none|xxs|xs|s|m|l|xl|xxl]
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .body_indent_$(value)
        .body__section
            @if $value == none
                padding-left: 0
                padding-right: 0
            @else
                padding-left: var(--ek-body-indent-$(value))
                padding-right: var(--ek-body-indent-$(value))
