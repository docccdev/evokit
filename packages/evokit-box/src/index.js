// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
