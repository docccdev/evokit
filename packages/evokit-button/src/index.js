import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
