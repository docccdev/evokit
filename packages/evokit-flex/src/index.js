import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'display',
    'content',
    'items',
    'direction',
    'wrap',
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'order',
]);

Flex.Item = FlexItem;
