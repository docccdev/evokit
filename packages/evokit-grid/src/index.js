import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
