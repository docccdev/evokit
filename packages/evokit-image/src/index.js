import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
