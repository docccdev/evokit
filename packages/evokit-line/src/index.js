import { createBlock } from 'evokit';

export const Line = createBlock('hr', 'line', [
    'display',
    'indent',
    'style',
    'color',
]);
