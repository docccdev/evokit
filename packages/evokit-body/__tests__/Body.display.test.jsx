import React from 'react';
import { shallow } from 'enzyme';
import { Body } from '../src';

describe('<Body body-display="..." />', () => {
    it('props body-display', () => {
        const wrapper = shallow(<Body body-display='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<body class="ek-body ek-body_display_value"></body>');
    });
});
