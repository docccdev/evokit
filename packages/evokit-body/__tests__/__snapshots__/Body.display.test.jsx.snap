// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body body-display="..." /> props body-display 1`] = `
<body
  className="ek-body ek-body_display_value"
/>
`;
